pkg/virt-api/rest/subresource_test.go

```diff
@@ -1208,7 +1209,57 @@ var _ = Describe("VirtualMachineInstance Subresources", func() {
 
 			statusErr := ExpectStatusErrorWithCode(recorder, http.StatusConflict)
 			// check the msg string that would be presented to virtctl output
-			Expect(statusErr.Error()).To(ContainSubstring("VM is not running"))
+			Expect(statusErr.Error()).To(ContainSubstring(fmt.Sprintf("VM has status %q and is not running", v1.Unknown)))
+		})
+
+		It("should fail on VM with RunStrategyHalted", func() {
```

Comment:
IIUC this is already covered in the table entry you wrote in line 1187 (`table.Entry("RunStrategyHalted", v1.RunStrategyHalted, "VM has no associated VMI running")`).

I also notice that here we expect for different error message (`"Halted does not support manual stop requests"`). How come? Am I missing something here?

Reply:
testing different things, one tests that no matter the run strategy if there isn't a VMI present we return an error not it is not running, one is testing that if a vmi is running but is using this strategy we return an error.